### PR TITLE
prelude: fix thread safeness

### PR DESCRIPTION
**Pull request by regit**
Prelude analyzer is not thread safe so we need to have one
analyzer per thread.

PR builds:
- PR build: https://buildbot.openinfosecfoundation.org/builders/regit/builds/127
- PR pcaps: https://buildbot.openinfosecfoundation.org/builders/regit-pcap/builds/125

Redmine tickets: https://redmine.openinfosecfoundation.org/issues/1639

### CODE REVIEW

File: src/alert-prelude.c [hunk 1]
```diff
@@ -618,6 +661,20 @@ static TmEcode AlertPreludeThreadInit(ThreadVars *t, void *initdata, void **data
 
     /** Use the Ouput Context */
     aun->ctx = ((OutputCtx *)initdata)->data;
+    
+    /** Create a per-thread idmef analyzer */
+    if (idmef_analyzer_new(&aun->analyzer) < 0) {
+        SCLogError(SC_ERR_INITIALIZATION,
+                   "Error creating idmef analyzer for Prelude.");
+        SCReturnInt(TM_ECODE_FAILED);
```
Comment (inliniac) [inline src/alert-prelude.c]: aun is lost here

File: src/alert-prelude.c [hunk 2]
```diff
@@ -618,6 +661,20 @@ static TmEcode AlertPreludeThreadInit(ThreadVars *t, void *initdata, void **data
...
+    }
+    
+    /** Setup the per-thread idmef analyzer */
+    if (SetupAnalyzer(aun->analyzer) < 0) {
+        SCLogError(SC_ERR_INITIALIZATION,
+                   "Error configuring idmef analyzer for Prelude.");
+        SCReturnInt(TM_ECODE_FAILED);
```
Comment (inliniac) [inline src/alert-prelude.c]: aun is lost

File: src/alert-prelude.c (hunk 1 above)
Comment (inliniac) [inline src/alert-prelude.c]: it's not a huge deal but I don't like adding output to my QA. Cppcheck says: [src/alert-prelude.c:676]: (error) Memory leak: aun

**Comment by regit** [comment]
We are working on addressing these issues.

**Comment by inliniac** [comment]
Merged this through #1800, added a fix for the memleak.

**Pull request closed without merging**